lib/pbench/cli/server/report.py

```diff
@@ -475,7 +475,7 @@ def report(
             cache_m = CacheManager(config, logger)
             verifier.status("starting discovery")
             watcher.update("discovering cache")
-            cache_m.full_discovery()
+            cache_m.full_discovery(full=False)
```

Comment:
Both varieties are "full" except in the presence of prior server errors. The new mode is faster and will ignore "orphaned" tarballs in the ARCHIVE tree that don't represent SQL `Dataset`s. The modern variation of the old "audit" tool would definitely want to locate and diagnose these, but in the context of my report generator the faster "full discover" is much more useful.

I suppose I could call it "paranoid" or "exhaustive" or just "slow", but I frankly didn't spend much time worrying about it and I'm not really inclined to start now.

Reply:
> Both varieties are "full"

OK, so that justifies keeping the function name, but it makes the keyword name look silly.

I would be inclined to replace `full` with `directory_search` or its opposite, `use_database`; or, perhaps, `by_controller` vs. `by_dataset`.

Reply:
I actually renamed the parameter to `search` (vs "by query", I guess).